src/solve.jl

```diff
@@ -100,8 +100,8 @@ function DiffEqBase.__init(prob::DiffEqBase.AbstractDDEProblem,
 
   progress && @logmsg(-1, progress_name, _id = _id = :DelayDiffEq, progress = 0)
 
-  isdae = prob.f.mass_matrix != I && !(typeof(prob.f.mass_matrix)<:Tuple) &&
-                                ArrayInterface.issingular(prob.f.mass_matrix)
+  isdae = prob.f.mass_matrix !== I && !(prob.f.mass_matrix isa Tuple) &&
```

Comment:
How does this fix the bug? Sorry, I just don't see the difference in logic. Is it because the equality check fails in some cases?

Reply:
The bug was that ArrayInterface was neither imported nor a dependency at all. I just reformatted the line while I was at it but there's no fix here.